packages/wallet/src/KeyManagement/LedgerKeyAgent.ts

```diff
@@ -0,0 +1,217 @@
+    this.#networkId = networkId;
+    this.#knownAddresses = knownAddresses;
+    this.#extendedAccountPublicKey = extendedAccountPublicKey;
+    this.#communicationType = communicationType;
+    this.deviceConnection = deviceConnection;
+  }
+
+  get networkId(): Cardano.NetworkId {
+    return this.#networkId;
+  }
+
+  get accountIndex(): number {
+    return this.#accountIndex;
+  }
+
+  get __typename(): KeyAgentType {
+    return KeyAgentType.Ledger;
+  }
+
+  get knownAddresses(): GroupedAddress[] {
+    return this.#knownAddresses;
+  }
+
+  get serializableData(): SerializableLedgerKeyAgentData {
+    return {
+      __typename: KeyAgentType.Ledger,
+      accountIndex: this.#accountIndex,
+      communicationType: this.#communicationType,
+      extendedAccountPublicKey: this.#extendedAccountPublicKey,
+      knownAddresses: this.#knownAddresses,
+      networkId: this.networkId
+    };
+  }
+
+  static async getHidDeviceList(): Promise<string[]> {
+    return await TransportNodeHid.list();
+  }
+
+  static async createTransport({
+    communicationType,
+    activeTransport,
+    devicePath = ''
+  }: CreateTransportProps): Promise<TransportType> {
+    if (communicationType === CommunicationType.Node) {
+      return await TransportNodeHid.open(devicePath);
+    }
+    return await (activeTransport && activeTransport instanceof TransportWebHID
+      ? TransportWebHID.open(activeTransport.device)
+      : TransportWebHID.request());
+  }
+
+  static async createDeviceConnection(activeTransport: Transport): Promise<AppAda> {
+    const deviceConnection = new AppAda(activeTransport);
+    // Perform app check to see if device can respond
+    await deviceConnection.getVersion();
+    return deviceConnection;
+  }
+
+  static async establishDeviceConnection(communicationType: CommunicationType, devicePath?: string): Promise<AppAda> {
+    let transport;
+    try {
+      transport = await LedgerKeyAgent.createTransport({ communicationType, devicePath });
+      if (!transport || !transport.deviceModel) {
+        throw new TransportError('Transport failed');
+      }
+      const isSupportedLedgerModel = transport.deviceModel.id === 'nanoS' || transport.deviceModel.id === 'nanoX';
+      if (!isSupportedLedgerModel) {
+        throw new TransportError('Ledger device model not supported');
+      }
+      return await LedgerKeyAgent.createDeviceConnection(transport);
+    } catch (error) {
+      if (error.message.includes('cannot open device with path')) {
+        throw new TransportError('Connection already established', error);
+      }
+      // If transport is established we need to close it so we can recover device from previous session
+      if (transport) {
+        // eslint-disable-next-line @typescript-eslint/no-floating-promises
+        transport.close();
+      }
+      throw error;
+    }
+  }
+
+  static async checkDeviceConnection(communicationType: CommunicationType, deviceConnection?: AppAda): Promise<AppAda> {
```

Comment:
:question: 
1. What triggers the application to use this method?
2. Is it defensive, or as a result on an error?
3. Can disconnection not be resolved by simply re-creating the class instance without this added logic?

Reply:
I think most of those static functions should be private

Reply:
It's still exposing internals of hardware device management, and I'd like to understand if we need to do that instead of just using abstract concepts and handling disconnections as an error state.